docs/design/2019-11-14-tidb-builtin-diagnostics-zh_CN.md

```diff
@@ -0,0 +1,680 @@
+| address    | varchar(32) | YES  |     | NULL              |       |
+| type       | varchar(32) | YES  |     | NULL              |       |
+| value      | float       | YES  |     | NULL              |       |
+| interval   | int         | YES  |     | 60                |       |
+| start_time | int         | YES  |     | CURRENT_TIMESTAMP |       |
++------------+-------------+------+-----+-------------------+-------+
+3 rows in set (0.02 sec)
+
+mysql> select address, type, value from pd_client_cmd_ops;
++------------------+----------------------+---------+
+| address          | type                 | value   |
++------------------+----------------------+---------+
+| 172.16.5.33:2379 | update_gc_safe_point |       0 |
+| 172.16.5.33:2379 | wait                 | 2.91052 |
+| 172.16.5.33:2379 | get_all_stores       |       0 |
+| 172.16.5.33:2379 | get_prev_region      |       0 |
+| 172.16.5.33:2379 | get_region           |       0 |
+| 172.16.5.33:2379 | get_region_byid      |       0 |
+| 172.16.5.33:2379 | scan_regions         |       0 |
+| 172.16.5.33:2379 | tso_async_wait       | 2.91052 |
+| 172.16.5.33:2379 | get_operator         |       0 |
+| 172.16.5.33:2379 | get_store            |       0 |
+| 172.16.5.33:2379 | scatter_region       |       0 |
++------------------+----------------------+---------+
+11 rows in set (0.00 sec)
+
+mysql> select address, type, value from pd_client_cmd_ops where start_time=’2019-11-14 10:00:00’;
++------------------+----------------------+---------+
+| address          | type                 | value   |
++------------------+----------------------+---------+
+| 172.16.5.33:2379 | update_gc_safe_point |       0 |
+| 172.16.5.33:2379 | wait                 | 0.82052 |
+| 172.16.5.33:2379 | get_all_stores       |       0 |
+| 172.16.5.33:2379 | get_prev_region      |       0 |
+| 172.16.5.33:2379 | get_region           |       0 |
+| 172.16.5.33:2379 | get_region_byid      |       0 |
+| 172.16.5.33:2379 | scan_regions         |       0 |
+| 172.16.5.33:2379 | tso_async_wait       | 0.82052 |
+| 172.16.5.33:2379 | get_operator         |       0 |
+| 172.16.5.33:2379 | get_store            |       0 |
+| 172.16.5.33:2379 | scatter_region       |       0 |
++------------------+----------------------+---------+
+11 rows in set (0.00 sec)
+```
+
+对于多个 label 的 PromQL 就会有多个列的数据，可以方便的使用已有的 SQL 执行引擎对数据过滤、聚合得到期望的结果。
+
+#### 节点性能剖析系统表
+
+通过各个节点的 `/debug/pprof/profile` 拿到对应节点性能采样数据，然后对采样数据进行聚合，最终使用 SQL 查询结果的方式想用户输出性能剖析结果。由于 SQL 查询结果不能以 svg 的格式输出，所以需要解决输出内容展示的问题。
+
+火焰图快速定位问题的核心点是：
+
+- 提供全局视野
+- 展示全部调用路径
+- 层次化展示
+
+本提案提出的解决方案聚焦在解决核心问题的点上，而未拘泥于是图形展示形式。最终的方案为：对采样数据进行聚合，并将所有的调用路径使用树形结构逐行进行展示。
+
+解决方案是通过以下方式契合三个核心点：
+
+- 提供全局视野：对每一个聚合结果使用单独的一列展示在全局的使用比例，可以方便过滤排序
+- 展示全部调用路径：将所有的调用路径都作为查询结果，并使用单独的列对各个调用路径的子树进行编号，可以方便的通过过滤只查看某一个子树
+- 层次化展示：使用树形结构展示堆栈，使用单独的列记录栈的深度，可以方便的对不同栈的深度进行过滤
+
+本提案需要实现以下性能剖析表：
+
+
+| 表名 | 作用 |
+|------|-----|
+| tidb_profile_cpu | TiDB CPU 火焰图 |
```

Comment:
maybe we can add a type for the perf table. 

Btw, we also need to see the PD perf sometimes.

Reply:
https://github.com/pingcap/tidb/pull/13717